### PR TITLE
fix: optimize router logics

### PR DESCRIPTION
**Pull request by Jeff-CCH**
Optimize router logics
- [x] Rename amountBps to bps or balanceBps
- [x] Move interfaces/I*FlashLoanCallback to interfaces/fees/
- [x] Merge external and internal set* functions in Router
- [x] Unnecessary return type: payable address
- [x] Unnecessary _INVALID_PAUSER and _INVALID_FEE_COLLECTOR constants
- [x] Move external setFeeCalculator(s) functions upward
- [x] Unnecessary _approveZero in MakerUtility::openLockGemAndDraw

### CODE REVIEW

File: src/fees/FeeGenerator.sol [hunk 1]
```diff
@@ -103,27 +124,6 @@ abstract contract FeeGenerator is Ownable {
...
-            setFeeCalculator(selector, to, feeCalculator);
-            unchecked {
-                ++i;
-            }
-        }
-    }
-
     function setFeeCalculator(bytes4 selector, address to, address feeCalculator) public onlyOwner {
         feeCalculators[selector][to] = feeCalculator;
         emit FeeCalculatorSet(selector, to, feeCalculator);
```
Comment (zodahu) [inline src/fees/FeeGenerator.sol:129]: Relocate this function to align with external `setFeeCalculators`

File: src/interfaces/fees/IAaveV2FlashLoanCallback.sol
Comment (zodahu) [inline src/interfaces/fees/IAaveV2FlashLoanCallback.sol:1]: Should be `interfaces/callbacks/`

File: src/fees/FeeGenerator.sol (hunk 1 above)
Comment (Jeff-CCH) [reply src/fees/FeeGenerator.sol]: fixed

File: src/interfaces/fees/IAaveV2FlashLoanCallback.sol
Comment (Jeff-CCH) [reply src/interfaces/fees/IAaveV2FlashLoanCallback.sol:1]: fixed

**Comment by zodahu** [comment]
LGTM

**Comment by MussinaLin** [comment]
LGTM

**Pull request merged**